lib/filewatch/watch.rb

```diff
@@ -117,127 +117,147 @@ def inode(path, stat)
+                # was first ignored we updated the bytes_read to the stat.size at that time.
+                # by adding this to the sincedb so that the subsequent modify
+                # event can detect the change
+                watched_file.watch
+                yield(:unignore, watched_file)
+              end
+            rescue Errno::ENOENT
+              # file has gone away or we can't read it anymore.
+              file_deletable << path
+              @logger.debug? && @logger.debug("each: ignored: stat failed: #{path}: (#{$!}), deleting from @files")
+            rescue => e
+              @logger.debug? && @logger.debug("each: ignored: stat failed: #{path}: (#{e.inspect})")
             end
-          rescue Errno::ENOENT
-            # file has gone away or we can't read it anymore.
-            file_deleteable << path
-            @logger.debug? && @logger.debug("each: ignored: stat failed: #{path}: (#{$!}), deleting from @files")
-          rescue => e
-            @logger.debug? && @logger.debug("each: ignored: stat failed: #{path}: (#{e.inspect})")
           end
-        end
 
-        # Send any creates.
-        if (to_take = @max_active - watched_files.count{|wf| wf.active?}) > 0
-          watched_files.select {|wf| wf.watched? }.take(to_take).each do |watched_file|
-            watched_file.activate
-            # don't do create again
-            next if watched_file.state_history_any?(:closed, :ignored)
-            # if the file can't be opened during the yield
-            # its state is set back to watched
-            if watched_file.initial?
-              yield(:create_initial, watched_file)
-            else
-              yield(:create, watched_file)
+          return if quit?
+
+          # Send any creates.
+          if (to_take = @max_active - watched_files.count{|wf| wf.active?}) > 0
+            watched_files.select {|wf| wf.watched? }.take(to_take).each do |watched_file|
+              break if quit?
+              path = watched_file.path
+              begin
+                stat = watched_file.restat
+                watched_file.activate
+                # don't do create again
+                next if watched_file.state_history_any?(:closed, :ignored)
+                # if the file can't be opened during the yield
+                # its state is set back to watched
+                sym = watched_file.initial? ? :create_initial : :create
+                yield(sym, watched_file)
+              rescue Errno::ENOENT
+                # file has gone away or we can't read it anymore.
+                file_deletable << path
+                watched_file.unwatch
+                yield(:delete, watched_file)
+                next
+                @logger.debug? && @logger.debug("each: closed: stat failed: #{path}: (#{$!}), deleting from @files")
+              rescue => e
+                @logger.debug? && @logger.debug("each: closed: stat failed: #{path}: (#{e.inspect})")
```

Comment:
Sorry, too hasty copy-pasta. Will fix the `"each: closed: stat failed:` in each case.
So yield(:create) will open a file and read it sending listener callbacks (created, accept) to the caller (file input)
Anything can happen.
So you are correct it should be `logger.error`.